src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs

```diff
@@ -235,27 +237,40 @@ internal void SetExpired(EvictionReason reason)
             DetachTokens();
         }
 
-        private bool CheckForExpiredTime(DateTimeOffset now)
+        private bool CheckForExpiredTime(in DateTimeOffset now)
         {
-            if (_absoluteExpiration.HasValue && _absoluteExpiration.Value <= now)
-            {
-                SetExpired(EvictionReason.Expired);
-                return true;
-            }
+            if (_absoluteExpiration == null && _absoluteExpiration == null)
```

Comment:
Is this the same check?

Reply:
@eerhardt do you mean if `_absoluteExpiration` and `_absoluteExpiration` are the same thing? they are not.

or whether the `nullable != null` is equal to `nullable.HasValue`? I've checked and it's compiled to the same code, but `.HasValue` looks better and leaves no questions so I am going to change it to `.HasValue`

<details>

```cs
using System;
public class C {
    DateTimeOffset? _absoluteExpiration;
    public int M() {
        if (_absoluteExpiration.HasValue)
            return 1;
        return 0;
    }
    public int N() {
        if (_absoluteExpiration != null)
            return 1;
        return 0;
    }
}
```

```asm
; Core CLR v4.700.20.20201 on x86

C..ctor()
    L0000: ret

C.M()
    L0000: add ecx, 4
    L0003: cmp byte ptr [ecx], 0
    L0006: je short L000e
    L0008: mov eax, 1
    L000d: ret
    L000e: xor eax, eax
    L0010: ret

C.N()
    L0000: add ecx, 4
    L0003: cmp byte ptr [ecx], 0
    L0006: je short L000e
    L0008: mov eax, 1
    L000d: ret
    L000e: xor eax, eax
    L0010: ret
```


</details>

Reply:
I guarantee that is not true.  😉